components/script/dom/element.rs

```diff
@@ -2656,7 +2739,7 @@ impl VirtualMethods for Element {
                         None
                     }
                 });
-                if node.is_in_doc() {
```

Comment:
This is wrong (or rather, incomplete). You don't want to register named elements in the document but in either the document or shadow root. You should be able to add tests for this by adding an element with `name` attribute into a shadowroot.